drivers/sensor/bosch/bme280/bme280_async.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2024 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+
+#include "bme280.h"
+
+LOG_MODULE_DECLARE(BME280, CONFIG_SENSOR_LOG_LEVEL);
+
+int bme280_submit(const struct device *dev, struct rtio_iodev_sqe *iodev_sqe)
+{
+	uint32_t min_buf_len = sizeof(struct bme280_encoded_data);
+	int rc;
+	uint8_t *buf;
+	uint32_t buf_len;
+	struct bme280_encoded_data *edata;
+
+	rc = rtio_sqe_rx_buf(iodev_sqe, min_buf_len, min_buf_len, &buf, &buf_len);
+	if (rc != 0) {
+		LOG_ERR("Failed to get a read buffer of size %u bytes", min_buf_len);
+		rtio_iodev_sqe_err(iodev_sqe, rc);
+		return rc;
+	}
+
+	edata = (struct bme280_encoded_data *)buf;
+	edata->header.timestamp = k_ticks_to_ns_floor64(k_uptime_ticks());
+
+	rc = bme280_sample_fetch_helper(dev, SENSOR_CHAN_ALL, &edata->data);
```

Comment:
Heads-up: This is a blocking call, similar to the ones referenced in #73676. Assuming the suggested fix goes in, this should be fine.

I'm assuming this will change to a non-blocking flow using RTIO bus drivers when we consider there's enough support to make the breaking change.

Reply:
yeah we *really* need to move forward with something here, I think after 3.7 some shim with k_work or p4wq would be useful to create a i2c/spi rtio shim so drivers can use the RTIO API against buses without having to worry about this. Its clear I think at this point we need it to move sensor drivers forward while we push on bus drivers to implement the API that would better suite the task.

Reply:
I'll attempt to capture an RFC for this

Reply:
Captured in #73869